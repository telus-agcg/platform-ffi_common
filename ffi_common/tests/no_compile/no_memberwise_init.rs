use ffi_common::derive::FFI;

#[derive(Debug, Clone, Copy, FFI)]
#[ffi(forbid_memberwise_init)]
pub struct NoInitializerAllowed {
    pub field1: u8,
}

fn test_foo() {
    // This should not compile because of the `forbid_memberwise_init` attribute.
    let _ = unsafe { no_initializer_allowed_ffi::no_initializer_allowed_rust_ffi_init(1) };
}

fn main() { }
