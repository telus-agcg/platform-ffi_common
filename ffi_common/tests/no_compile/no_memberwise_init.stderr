error[E0425]: cannot find function `no_initializer_allowed_rust_ffi_init` in module `no_initializer_allowed_ffi`
  --> tests/no_compile/no_memberwise_init.rs:11:50
   |
3  | #[derive(Debug, Clone, Copy, FFI)]
   |                              --- similarly named function `no_initializer_allowed_rust_ffi_free` defined here
...
11 |     let _ = unsafe { no_initializer_allowed_ffi::no_initializer_allowed_rust_ffi_init(1) };
   |                                                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: a function with a similar name exists: `no_initializer_allowed_rust_ffi_free`
